src/components/UserInfo.js

```diff
@@ -1,18 +1,53 @@
-export default class userInfo {
-  constructor({ nameSelector, infoSelector }) {
+export default class UserInfo {
+  constructor({ nameSelector, infoSelector, avatarSelector }) {
     this.profileTitle = document.querySelector(nameSelector);
     this.profileDescription = document.querySelector(infoSelector);
+    this.profileAvatar = document.querySelector(avatarSelector);
   }
 
+  // Fetch current user info
   getUserInfo() {
     return {
       title: this.profileTitle.textContent,
       description: this.profileDescription.textContent,
     };
   }
 
+  // Fetch user info from the server
+  fetchUserInfo() {
+    if (!this._baseUrl || !this._headers) {
+      console.error("Base URL or headers are not set.");
+      return Promise.reject(new Error("Base URL or headers are missing"));
+    }
+
+    console.log("Fetching user info...");
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    })
+      .then(this._checkResponse)
+      .catch(this._handleError);
```

Comment:
Please, delete `fetchUserInfo` from `UserInfo` because you don't use it. And you can't use it from here anyway